### PR TITLE
Rawg add and errMsg function 

### PR DESCRIPTION
**Pull request by ryanchiu2017**

### CODE REVIEW

File: functions/command/add.go [hunk 1]
```diff
@@ -30,7 +32,37 @@ func (c *AddCommand) Execute(ctx context.Context, cl *clients.Clients) (*discord
...
+		typ = activity.GAME
+
+		ctx, cancel := context.WithTimeout(context.Background(), time.Duration(time.Millisecond*500))
+		defer cancel()
+
+		detail, err := cl.Rawg().GetGame(ctx, c.Name)
+
+		if err != nil {
+			msg := clients.RawgErrorMsg(err)
+			return utils.NewWebhookEdit(msg), err
```
Comment (PinkNoize) [inline functions/command/add.go:44]: Should return a nil error when the game is not found. That is a user error and not an error on our end

File: functions/command/add.go [hunk 2]
```diff
@@ -30,7 +32,37 @@ func (c *AddCommand) Execute(ctx context.Context, cl *clients.Clients) (*discord
...
+		if err != nil {
+			ae, ok := err.(*activity.ActivityError)
+			if ok {
+				if ae.Reason == activity.ALREADY_EXISTS {
+					return utils.NewWebhookEdit(fmt.Sprintf("%v already exists in the pool", c.Name)), nil
+				}
+			}
+			return nil, fmt.Errorf("act.Create: %v", err)
+		}
+		return utils.NewWebhookEdit(fmt.Sprintf("%v added to the pool", c.Name)), nil
```
Comment (PinkNoize) [inline functions/command/add.go:65]: Don't duplicate the code below. Try to set only the required variables in this branch and use the create outside of the switch below.

File: functions/clients/rawg.go [hunk 3]
```diff
@@ -24,3 +24,19 @@ func NewRawg(rawgToken string) *Rawg {
...
+		case http.StatusNotFound:
+			return "Bot not connected to servers at the moment"
+		case http.StatusUnauthorized:
+			return "RAWG API not availible at the moment"
+		default:
+			return "Unspecified HTTP error occoured"
+		}
+	} else {
+		return "Unspecified error occoured"
+	}
```
Comment (PinkNoize) [inline functions/clients/rawg.go:41]: Messages don't match the error code. I'm also not sure we should show the error to the user.

File: functions/clients/rawg.go [hunk 4]
```diff
@@ -24,3 +24,19 @@ func NewRawg(rawgToken string) *Rawg {
 func (r *Rawg) GetGame(ctx context.Context, game string) (*rawg.GameDetailed, error) {
 	return r.client.GetGame(ctx, game)
 }
+
+func RawgErrorMsg(err error) string {
+	rawgError, ok := err.(*rawg.RawgError)
```
Comment (PinkNoize) [inline functions/clients/rawg.go:29]: Not sure I like converting an error to a string. We might not need this function as adding a game will likely be the only place we check for a 404.

**Review by PinkNoize — CHANGES_REQUESTED** [review]

File: functions/clients/rawg.go [hunk 5]
```diff
@@ -22,5 +23,6 @@ func NewRawg(rawgToken string) *Rawg {
 }
 
 func (r *Rawg) GetGame(ctx context.Context, game string) (*rawg.GameDetailed, error) {
-	return r.client.GetGame(ctx, game)
+	detail, err := r.client.GetGame(ctx, game)
+	return detail, fmt.Errorf("rawg.go: %v", err)
```
Comment (PinkNoize) [inline functions/clients/rawg.go:27]: Leave this as it was. Returning errorf will prevent it from being converted into a more specific error. If this function gets more complicated later we can do this with ErrorW (I think)

File: functions/command/add.go [hunk 6]
```diff
@@ -26,15 +30,38 @@ func NewAddCommand(guildID, activityType, name string) *AddCommand {
...
+		ctx, cancel := context.WithTimeout(context.Background(), time.Duration(time.Millisecond*500))
+		defer cancel()
+
+		detail, err := cl.Rawg().GetGame(ctx, c.Name)
+
+		if err != nil {
+			if rawgError, ok := err.(*rawg.RawgError); ok && rawgError.HttpCode == http.StatusNotFound {
+				return utils.NewWebhookEdit("🚧 Game not found 🚧"), err
+			} else {
+				return utils.NewWebhookEdit("🤖🔥 Bot error 🔥"), err
```
Comment (PinkNoize) [inline functions/command/add.go:49]:
```suggestion
				return nil, err
```
We have a default error in the Command function handler code

File: functions/command/add.go [hunk 7]
```diff
@@ -26,15 +30,38 @@ func NewAddCommand(guildID, activityType, name string) *AddCommand {
...
+			}
+		}
+		if detail == nil {
+			return utils.NewWebhookEdit("🚧 Game not found 🚧"), err
+		}
+
+		info = &activity.GameInfo{
+			Id:              strconv.Itoa(detail.ID),
```
Comment (PinkNoize) [inline functions/command/add.go:57]: Go ahead and change the GameInfo id type to an int so you don't have to convert it here

File: functions/command/add.go [hunk 8]
```diff
@@ -26,15 +30,38 @@ func NewAddCommand(guildID, activityType, name string) *AddCommand {
...
 	var typ activity.ActivityType
+	var info *activity.GameInfo
 	switch c.ActivityType {
 	case "activity":
 		typ = activity.ACTIVITY
 	case "game":
-		return utils.NewWebhookEdit("🚧 Games not implemented yet"), nil
+		typ = activity.GAME
+
+		ctx, cancel := context.WithTimeout(context.Background(), time.Duration(time.Millisecond*500))
```
Comment (PinkNoize) [inline functions/command/add.go:40]: Using the existing ctx instead of getting the background one. Is there a reason this needs to complete in 500 ms?

**Review by PinkNoize — CHANGES_REQUESTED** [review]

File: functions/command/add.go (hunk 8 above)
Comment (ryanchiu2017) [reply functions/command/add.go:40]: it should be fine without it, it was just from the example code

File: functions/clients/rawg.go [hunk 9]
```diff
@@ -22,5 +22,5 @@ func NewRawg(rawgToken string) *Rawg {
 }
 
 func (r *Rawg) GetGame(ctx context.Context, game string) (*rawg.GameDetailed, error) {
-	return r.client.GetGame(ctx, game)
+	return detail, err
```
Comment (PinkNoize) [inline functions/clients/rawg.go:25]: Fix this

**Review by PinkNoize — CHANGES_REQUESTED** [review]

File: functions/clients/rawg.go (hunk 9 above)
Comment (ryanchiu2017) [reply functions/clients/rawg.go:25]: oops

**Review by PinkNoize — APPROVED** [review]

**Pull request merged**